### PR TITLE
CI: Expand E4S ROCm stack to include missing DaV packages

### PR DESCRIPTION
**Pull request by kwryankrattiger**
Ascent: Fixup for VTK-m with Kokkos backend

### CODE REVIEW

File: var/spack/repos/builtin/packages/ascent/package.py [hunk 1]
```diff
@@ -181,6 +181,8 @@ class Ascent(CMakePackage, CudaPackage):
     depends_on("vtk-m+cuda", when="@0.9.0: +vtkh+cuda")
     depends_on("vtk-m+fpic", when="@0.8.0: +vtkh")
     depends_on("vtk-m~shared+fpic", when="@0.8.0: +vtkh~shared")
+    # Ascent defaults to C++11
+    depends_on("kokkos std=11", when="+vtkh ^vtk-m +kokkos")
```
Comment (kwryankrattiger) [inline var/spack/repos/builtin/packages/ascent/package.py:187]: Found this when building Ascent with ROCm enabled VTK-m.

File: share/spack/gitlab/cloud_pipelines/stacks/e4s/spack.yaml [hunk 2]
```diff
@@ -189,7 +189,7 @@ spack:
   - cabana +rocm
   - caliper +rocm
   - chai ~benchmarks +rocm
-  - ecp-data-vis-sdk +paraview +vtkm +rocm
+  - ecp-data-vis-sdk +ascent +adios2 +hdf5 +paraview +pnetcdf +sz +vtkm +zfp +rocm
```
Comment (kwryankrattiger) [inline share/spack/gitlab/cloud_pipelines/stacks/e4s/spack.yaml]: Only ascent, paraview, and vtk-m have ROCm support, the other packages are integrations that are useful to have in conjunction with the ROCm enabled packages.

**Review by cyrush — DISMISSED** [review]

**Comment by cyrush** [comment]
FYI: I still haven't been able to add ROCm logic for ascent 0.9.0 to spack yet.

**Review by eugeneswalker — DISMISSED** [review]
Thank you!

**Review by eugeneswalker — DISMISSED** [review]

**Comment by kwryankrattiger** [comment]
@spackbot run pipeline

**Comment by kwryankrattiger** [comment]
newer Kokkos doesn't work with c++11. I didn't exhaustively test which versions this is true in, but should be enough for now.
https://gitlab.spack.io/spack/spack/-/jobs/6926417

**Comment by kwryankrattiger** [comment]
Judging from this page, the 3.7 cut-off should be correct, but there isn't anything explicit in the docs.

https://kokkos.github.io/kokkos-core-wiki/deprecation_page.html

**Comment by kwryankrattiger** [comment]
@eugeneswalker I think this is pretty close to done. Of all of the patches for the next release this is the most important.

**Review by eugeneswalker — DISMISSED** [review]
Thanks!

**Comment by kwryankrattiger** [comment]
@cyrush dropping cuda ascent again
https://gitlab.spack.io/spack/spack/-/jobs/6990724

**Review by eugeneswalker — APPROVED** [review]

**Comment by cyrush** [comment]
@kwryankrattiger Thanks for the heads up. Looks like something is crossed with the way cuda is being presented, spack develop has diverged quite a bit from the spack we still use for our CI where we are successful with cuda.

We have been relying on non spack scripts, not sure when we will have the time to heal spack develop.

**Pull request merged**